src/test/test_step.go

```diff
@@ -347,14 +347,14 @@ func testCommandAndEnv(state *core.BuildState, target *core.BuildTarget, run int
 	return replacedCmd, env, err
 }
 
-func runTest(state *core.BuildState, target *core.BuildTarget, run int) ([]byte, error) {
+func runTest(state *core.BuildState, target *core.BuildTarget, run int) ([]byte, []byte, error) {
 	replacedCmd, env, err := testCommandAndEnv(state, target, run)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	log.Debugf("Running test %s#%d\nENVIRONMENT:\n%s\n%s", target.Label, run, strings.Join(env, "\n"), replacedCmd)
-	_, stderr, err := state.ProcessExecutor.ExecWithTimeoutShellStdStreams(target, target.TestDir(run), env, target.Test.Timeout, state.ShowAllOutput, false, process.NewSandboxConfig(target.Test.Sandbox, target.Test.Sandbox), replacedCmd, state.DebugFailingTests)
```

Comment:
I think `stderr` is badly named here. It looks like it should contain all the output from both stdout and stderr, so this should've worked fine.